src/services/auth.js

```diff
@@ -48,6 +49,20 @@ passport.use(new HTTPSignatureStrategy(
       })
   }))
 
+passport.use(new ClientCertStrategy((userName, info, done) => {
+  if (!config.auth.client_certificates_enabled) {
+    return done(new UnauthorizedError('Unsupported authentication method'))
+  }
+
+  Account.findByName(userName)
+    .then(function (userObj) {
+      if (!userObj || userObj.is_disabled || userObj.fingerprint !== info.fingerprint) {
```

Comment:
> using the fingerprint itself as the username

Would this mean that other authentication methods would be removed? Or some usernames are fingerprints and some aren't?

> the Common Name as the username

Do you mean putting the username in the common name field? If so, we should update the spec to restrict usernames to 64 characters: http://security.stackexchange.com/questions/40026/openssl-x509-whats-the-significance-of-cn-common-name